src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/Tree.java

```diff
@@ -276,6 +307,41 @@ private static boolean hasEnoughLeaves(@NotNull final IBlockAccess world, final
     }
 
     /**
+     * Check if the Lj is supposed to cut a tree.
+     * @param world the world it is in.
+     * @param pos the position a leaf is at.
+     * @param treesToCut the trees he is supposed to cut.
+     * @param leafPos
+     * @return false if not.
+     */
+    private static boolean supposedToCut(final IBlockAccess world, final BlockPos pos, final Map<ItemStorage, Boolean> treesToCut, final BlockPos leafPos)
+    {
+        final IBlockState state = world.getBlockState(pos);
+
+        for(final ItemStorage stack: treesToCut.entrySet().stream().filter(entry -> !entry.getValue()).map(Map.Entry::getKey).collect(Collectors.toList()))
+        {
+            final int variantNumber = calcVariantNumber(state.getBlock(), pos, leafPos, world);
+
+            if (Compatibility.isSlimeLeaf(world.getBlockState(leafPos).getBlock()))
+            {
+                if(Compatibility.isSlimeSapling(((ItemBlock) stack.getStack()).getBlock()) && variantNumber == stack.getItemStack().getMetadata())
+                {
+                    return false;
+                }
+            }
+            else if(!Compatibility.isSlimeSapling(((ItemBlock) stack.getStack()).getBlock()))
+            {
+                if(variantNumber == stack.getItemStack().getMetadata())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)